codice/ddf
# DDF-1771 Implement FederationAdmin bean for storing/editing federation configurations

#### What does this PR do?
This introduces the registry's Federation Admin Service. 
FederationAdminServiceImpl will come up on init and check existing registry entries for the identity node. If one doesn't exist, it will create itself
There was also some minor refactoring in the registry.
#### Who is reviewing it?
@clockard @vinamartin @brianfelix @oscaritoro @mcalcote @ricklarsen @stustison 
#### How should this be tested?
Install catalog-registry feature
Using solr query the catalog and verify an entry is displayed with the registry-node-status set to "PRIMARY"
#### Any background context you want to provide?
#### What are the relevant tickets?
DDF-1771
#### Screenshots (if appropriate)
#### Checklist:

  Adding the following bundles to the catalog-registry feature:
    catalog-registry-api
    catalog-registry-schema-bindings
    catalog-registry-federationadminservice
    catalog-registry-federationadminserviceimpl

Moving RegistryObjectMetacardType to catalog-registry-api
Updating RegistryPackageConverter and RegistryTransformer for the refactor
Updating IdentificationPluging for the refactor
Updating RegistryPolicyPlugin for the refactor

Bundle-izing catalog-registry-schema-bindings
Updating poms that were embedding it

Adding catalog-registry-api and supporting files
Adding catalog-registry-federationadminservice & impl and supporting files
FederationAdminServiceImpl will come up on init and check existing registry entries for the identity node. If one doesn't exist, it will create itself.
